src/decimal.rs

```diff
@@ -1694,8 +1694,68 @@ const fn flags(neg: bool, scale: u32) -> u32 {
     (scale << SCALE_SHIFT) | ((neg as u32) << SIGN_SHIFT)
 }
 
+macro_rules! impl_try_from_decimal {
+    ($TInto:ty, $conversion_fn:path, $name:expr) => {
+        ///
+        /// Try to convert a `Decimal` to a primitive type.
```

Comment:
I wonder if you could leverage `doc` macros here - similar to `arithmetic_impl.rs`. e.g.:

```
#[doc = concat!(
            "Try to convert a `Decimal` into `",
            stringify!($TInto),
            "`. Returns an error if the `Decimal` is unable to be represented as a `",
            stringify!($TInto),
            "`."
        )]
```

You could probably extend it to also reference the integer logic too (e.g. passed through as `$additional_doc:expr` or something).